c7n/mu.py

```diff
@@ -395,7 +398,21 @@ def publish(self, func, alias=None, role=None, s3_uri=None):
         else:
             func.alias = func.arn
 
-        for e in func.get_events(self.session_factory):
+        # process any previous lambda configuration
+        old_group_name = None
+        if existing:
```

Comment:
this logic should really be in the event implementation, rather than hardcoding a particular event's pattern into the manager. the notion is to pass both current, and previous of the function so the event can do whatever specific logic it needs. as is this just violates encapsulation.

Reply:
I'll work on it.